scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,559 @@
+
+    References
+    ----------
+    .. [1] Tsallis C (1988). "Possible generalization of Boltzmann-Gibbs
+        statistics." Journal of Statistical Physics, 52, 479-487.
+    .. [2] Tsallis C, Stariolo DA (1996). "Generalized Simulated Annealing."
+        Physica A, 233, 395-406.
+    .. [3] Xiang Y, Sun DY, Fan W, Gong XG (1997). "Generalized Simulated
+        Annealing Algorithm and Its Application to the Thomson Model."
+        Physics Letters A, 233, 216-220.
+    .. [4] Xiang Y, Gong XG (2000a). "Efficiency of Generalized Simulated
+        Annealing." PHYSICAL REVIEW E, 62, 4473.
+    .. [5] Xiang Y, Gubian S, Suomela B, Hoeng J. (2013). "Generalized
+        Simulated Annealing for Efficient Global Optimization: the GenSA
+        Package for R". The R Journal, Volume 5/1, June 2013.
+        http://journal.r-project.org/.
+    .. [6] Mullen, K. (2014). Continuous Global Optimization in R. Journal of
+        Statistical Software, 60(6), 1 - 45.
+        http://dx.doi.org/10.18637/jss.v060.i06
+    .. [7] Xiang Y, Gubian S, Martin F. Simulated dual annealing method for
+        efficient global optimization. Forthcoming.
+
+    Examples
+    --------
+    The following example is a 10-dimensional problem, with many local minima.
+    The function involved is called Rastrigin
+    (https://en.wikipedia.org/wiki/Rastrigin_function)
+
+    >>> from scipy.optimize import simulated_annealing
+    >>> func = lambda x: np.sum(x * x - 10 * np.cos(
+    ...    2 * np.pi * x)) + 10 * np.size(x)
+    >>> lw = [-5.12] * 10
+    >>> up = [5.12] * 10
+    >>> ret = simulated_annealing(func, None, bounds=list(zip(lw, up)))
+    >>> print("global minimum: xmin = {0}, f(xmin) = {1}".format(
+    ...    ret.x, ret.fun))
+    """
+
+    if x0 is not None and not len(x0) == len(bounds):
+        raise ValueError('Bounds size does not match x0')
+    lu = list(zip(*bounds))
+    lower = np.array(lu[0])
+    upper = np.array(lu[1])
+    # Checking that bounds are consistent
+    if not np.all(lower < upper):
+        raise ValueError('Bounds are note consistent min < max')
+    # Wrapper for the objective function and minimizer
+    obj_fun_wrapper = ObjectiveFunWrapper(bounds, func, **local_search_options)
+    if not local_search_options:
+        obj_fun_wrapper.set_default_minimizer()
+    # Initialization of RandomState for reproducible runs if seed provided
+    rand_state = check_random_state(seed)
+    # Initialization of the energy state
+    energy_state = EnergyState(lower, upper)
+    energy_state.reset(obj_fun_wrapper, rand_state, x0)
+    # Minimum value of annealing temperature reached to perform
+    # re-annealing
+    temperature_restart = 0.1
+    # VisitingDistribution instance
+    visit_dist = VisitingDistribution(lower, upper, visit, rand_state)
+    # Markov chain instance
+    markov_chain = MarkovChain(accept, visit_dist, obj_fun_wrapper, rand_state,
+                               energy_state)
+
+    # Run the search lopp 
+
```

Comment:
Why do you break the line here? I do not see any need for it

Reply:
Removed.